### PR TITLE
[Feature]implement bitmap_subset_limit function

### LINKED ISSUE #18448: 麻烦实现bitmap_subset_limit函数和min_by函数
麻烦实现 min_by函数。 为什么实现了max_by不实现min_by？  
参考 https://doris.apache.org/zh-CN/docs/dev/sql-manual/sql-functions/aggregate-functions/min_by

麻烦实现bitmap_subset_limit函数。  为什么实现了sub_bitmap不实现bitmap_subset_limit？  
参考 https://doris.apache.org/zh-CN/docs/dev/sql-manual/sql-functions/bitmap-functions/bitmap_subset_limit

函数不够丰富，用起来太难受了。

### PR DESCRIPTION
**Pull request by ucasfl**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #18448.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Review by Seaven — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by Seaven** [comment]
run starrocks_be_unittest

**Comment by Seaven** [comment]
run all

**Review by Seaven — DISMISSED** [review]

File: be/src/exprs/bitmap_functions.cpp [hunk 1]
```diff
@@ -569,4 +561,46 @@ StatusOr<ColumnPtr> BitmapFunctions::bitmap_to_base64(FunctionContext* context,
...
+        auto bitmap = bitmap_viewer.value(row);
+        auto range_start = range_start_viewer.value(row);
+        auto limit = limit_viewer.value(row);
+
+        if (range_start < 0) {
+            range_start = 0;
+        }
+
+        if (bitmap->cardinality() == 0) {
+            builder.append_null();
```
Comment (fzhedu) [inline be/src/exprs/bitmap_functions.cpp:590]: empty bitmap returns null by design?

File: be/src/types/bitmap_value.cpp [hunk 2]
```diff
@@ -969,4 +969,50 @@ int64_t BitmapValue::sub_bitmap_internal(const int64_t& offset, const int64_t& l
...
+    if (limit < 0) {
+        is_reverse = true;
+        abs_limit = -limit;
+    }
+
+    int64_t count = 0;
+
+    if (is_reverse) {
+        detail::Roaring64Map::const_iterator start = _bitmap->begin();
+        detail::Roaring64Map::const_iterator end = _bitmap->begin();
```
Comment (fzhedu) [inline be/src/types/bitmap_value.cpp:987]: why not use rbegin() and rend()? that will do like not-reverse case.

**Review by fzhedu — COMMENTED** [review]
pls add doc

File: be/src/exprs/bitmap_functions.cpp (hunk 1 above)
Comment (ucasfl) [reply be/src/exprs/bitmap_functions.cpp:590]: Yes.

**Review by Seaven — APPROVED** [review]
LGTM, wait the docment enginer review plz

**Review by stdpain — APPROVED** [review]

**Comment by stdpain** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**